tools/manifest/manifest.py

```diff
@@ -113,15 +113,20 @@ def __init__(self, tests_root: Text, url_base: Text = "/") -> None:
         self.url_base: Text = url_base
 
     def __iter__(self) -> Iterator[Tuple[Text, Text, Set[ManifestItem]]]:
-        return self.itertypes()
-
-    def itertypes(self, *types: Text) -> Iterator[Tuple[Text, Text, Set[ManifestItem]]]:
-        for item_type in (types or sorted(self._data.keys())):
+        for item_type in sorted(self._data):
             for path in self._data[item_type]:
                 rel_path = os.sep.join(path)
                 tests = self._data[item_type][path]
                 yield item_type, rel_path, tests
 
+    def itertypes(self, *types: Text) -> "Manifest":
```

Comment:
Should we give this function a different name now? This now does not return an iterable, but instead a filtered_manifest.

Reply:
The `Manifest` _is_ an `Iterable[Tuple[str, str, Set[ManifestItem]]]` (see `__iter__` above). This is meant to be backwards-compatible when the caller iterates `itertypes(...)` without explicitly invoking `next(...)` (i.e., almost always).

(The `filtered_manifest` initialization is admittedly kind of ugly, but this can be refactored later.)

Reply:
I found a better way: Simply don't load irrelevant test types in the first place. There's already some support for this; this has been extended by plumbing `--test-types -> testloader.ManifestLoader(types=...) -> manifest.load_and_update(types=...)` and dropping irrelevant types after the update. This also unifies the behavior of `load_and_update(types=...)` for the update/no-update cases.